### PR TITLE
feat/color icon command

### PR DESCRIPTION
**Pull request by hilmoo**
I haven't tested it yet because it requires a level 2 Discord Nitro server. It might also need a bit of enhancement since, for now, it only stores the bytes to the command.

- [x] testing

### CODE REVIEW

File: bot/cogs/color.py [hunk 1]
```diff
@@ -385,6 +387,42 @@ async def set_color(
...
+    @app_commands.describe(
+        role='The role you want to attach the icon.',
+        icon='The icon you want to attach to the role. Must be a PNG or JPEG(JPG included) file (Maximum 256 KB).',
+    )
+    async def set_icon(
+        self, interaction: Interaction, role: discord.Role, icon: discord.Attachment
+    ) -> None:
+        await interaction.response.defer()
+        if role not in interaction.user.roles:
+            return await interaction.followup.send("❌ Please use the role first.", ephemeral=True)
```
Comment (Iqrar99) [inline bot/cogs/color.py:400]: This is an efficient way to check the role
https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.get_role

File: bot/cogs/color.py [hunk 2]
```diff
@@ -385,6 +387,42 @@ async def set_color(
...
+            return await interaction.followup.send(
+                "❌ The image file size must be less than 256 KB!", ephemeral=True
+            )
+
+        byte = await icon.read()
+        bytes = io.BytesIO(byte)
+        file = discord.File(bytes, icon.filename)
+
+        embed = discord.Embed(
+            description=f"Please ask admin or mod to attach this icon to {role.mention}.\nThe button will be disabled in 3 hours.",
```
Comment (Iqrar99) [inline bot/cogs/color.py:417]: No need to be that long. 1 hour is enough

File: bot/cogs/color.py (hunk 1 above)
Comment (hilmoo) [reply bot/cogs/color.py:400]: let me check

File: bot/cogs/color.py (hunk 2 above)
Comment (hilmoo) [reply bot/cogs/color.py:417]: okay

**Comment by hilmoo** [comment]
tested, lgtm
https://discord.com/channels/761644411486339073/774322083319775262/1269284083264131092
image

File: bot/cogs/color.py [hunk 3]
```diff
@@ -402,6 +404,42 @@ async def set_color(
             )
             return await interaction.followup.send(embed=embed)
 
+    @app_commands.command(name='icon', description='Attach a icon on custom role.')
```
Comment (Iqrar99) [inline bot/cogs/color.py:407]:
```suggestion
    @app_commands.command(name='icon', description='Attach an icon on custom role.')
```

File: bot/cogs/views/color.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class AcceptIconAttachment(discord.ui.View):
+    def __init__(self, role, bytes):
+        super().__init__(timeout=3600)
+        self.role: discord.Role = role
+        self.bytes: BytesIO = bytes
+
+    @discord.ui.button(label="Accept", style=discord.ButtonStyle.success, emoji="✅")
+    async def add_role_icon(self, interaction: Interaction, button: discord.ui.Button):
```
Comment (Iqrar99) [inline bot/cogs/views/color.py:16]: Add type hint on expected return

**Review by Iqrar99 — CHANGES_REQUESTED** [review]

**Review by Iqrar99 — APPROVED** [review]
LGTM

**Pull request merged**